### PR TITLE
New Lobby: !help and $help commands

### PR DESCRIPTION
**Pull request by dusekj4**
**Issue**
#157 

**Test**
1. go to >>Library >> Commands

### CODE REVIEW

**Comment by Jazcash** [comment]
As for the general approach, I think this is the best we can do for now. Sending direct messages to bots and parsing the response obviously feels hacky, but there's not much else we can do about it unless server/SPADS provide a proper commend reference API. Parsing from their github docs is also an option, but I think I prefer this way for now because there's no guarantee their docs will be kept up to date.

I'm not sure the dropdown for filtering by name _or_ description is necessary, because I don't think sorting by description adds much value.

No need to worry about styling stuff too much, what you've done looks fine for now. I'll probably be iterating on style/design over time anyways.

I have some ideas for improvement, but they're more like additional feature requests so I'll open them as new issues after this is merged.

File: src/renderer/views/library/commands.vue [hunk 1]
```diff
@@ -3,20 +3,126 @@
...
+    api.comms.request("c.communication.send_direct_message", {
+        recipient_id: 3137,
+        message: `$help`,
+    });
+});
+
+// Unsubscribe from the response listener when the component is not visible anymore
+onUnmounted(() => {
+    /**
+  I don't know how to unsubscribe from the response listener
```
Comment (Jazcash) [inline src/renderer/views/library/commands.vue:94]: `.onResponse` returns a `SignalBinding` instance that you can later call `.destroy()` on to solve this. See `HostBattle.vue` for an example

File: src/renderer/views/library/commands.vue [hunk 2]
```diff
@@ -3,20 +3,126 @@
...
+
+        cmdDescription && !cmdDescription.includes("*") && commands.push({ cmd, cmdDescription });
+    });
+    // Send a message to the server to get all the commands
+    api.comms.request("c.communication.send_direct_message", {
+        recipient_id: 3137,
+        message: `!helpall`,
+    });
+    api.comms.request("c.communication.send_direct_message", {
+        recipient_id: 3137,
```
Comment (Jazcash) [inline src/renderer/views/library/commands.vue:86]: Does this work? I thought `$help` only worked in public chat or in private message with `Coordinator`?

Comment (Jazcash) [reply src/renderer/views/library/commands.vue:86]: Yeah that's what I meant. I know `!help` works but I would've been surprised if `$help` works. `!`are SPADS commands for the autohosts, and `$` are server commands.

Default SPADS commands: http://planetspads.free.fr/spads/doc/spadsDoc.html (https://github.com/Yaribz/SPADS/blob/master/doc/spadsDoc_Commands.html)
Server commands: https://github.com/beyond-all-reason/teiserver/blob/9fa121b0a7e011957f90aad153c77badca593e11/lib/teiserver/coordinator/coordinator_lib.ex

Basically just change this `recipient_id` for `$help` to `3082` and it should work.
